server/server.go

```diff
@@ -230,11 +233,10 @@ func NewWithMetrics(cfg Config, metrics *Metrics) (*Server, error) {
 }
 
 func newServer(cfg Config, metrics *Metrics) (*Server, error) {
-	// If user doesn't supply a logging implementation, by default instantiate
-	// logrus.
+	// If user doesn't supply a logging implementation, by default instantiate go-kit.
 	logger := cfg.Log
 	if logger == nil {
-		logger = log.NewLogrus(cfg.LogLevel)
+		logger = log.NewGoKit(log.LogfmtFormat, os.Stderr)
```

Comment:
This is a regression. I think you misunderstood my previous comment. You should also have a `log. NewGoKitWithLevel()` where you pass the level and the level filter it added to the logger. Then you call it here, passing `cfg.LogLevel`.